HerPublicWebsite.UnitTests/BusinessLogic/Services/RegularJobsServiceTests.cs

```diff
@@ -180,4 +185,38 @@ public async Task RunNightlyTasksAsync_WhenCalledWritingTheSecondFileFails_Updat
         allReferralListForCustodianCode6.Should().AllSatisfy(rr => rr.ReferralWrittenToCsv.Should().BeFalse());
         mockDataAccessProvider.Verify(dap => dap.PersistAllChangesAsync());
     }
+
+    [Test]
+    public async Task AddWorkingDaysToDateTime_WhenCalledOnADayFollowingABankHoliday_ReturnsThoseReferrals()
+    {
+        // Arrange
+        DateTime initialDateTime = new DateTime(2023, 03, 23);
+        mockHttpHandler.Expect("https://www.gov.uk/bank-holidays.json")
+            .Respond("application/json", @"{
+  'england-and-wales': {
+    'division': 'england-and-wales',
+    'events': [
+        {
+            'title':'New Year’s Day',
+            'date':'2023-01-01',
+        },
+        {
+            'title':'Fake Bank Holiday on Weekday',
+            'date':'2023-03-20',
+        },
+        {
+            'title':'Fake Holiday on Weekend',
+            'date':'2023-03-19',
+        }
+        ]
+    }
+}"
```

Comment:
Indent the JSON in the same manner as ths code:
```
        mockHttpHandler.Expect("https://www.gov.uk/bank-holidays.json")
            .Respond("application/json", @"{
                'england-and-wales': {
                    'division': 'england-and-wales',
                    'events': [
                        {
                            'title': 'New Year’s Day',
                            'date': '2023-01-01',
                        },
                        {
                            'title': 'Fake Bank Holiday on Weekday',
                            'date': '2023-03-20',
                        },
                        {
                            'title': 'Fake Holiday on Weekend',
                            'date': '2023-03-19',
                        }
                    ]
                }
            }");
```

Reply:
When moving this out into a separate test file (when you make the extensions refactor), I recommend doing this mocking in the `SetUp()` method.

However, I'd pull out the mocked JSON into a separate static function to reduce clutter.

I would also suggest that you just use the real set of bank holidays, rather than faked ones. May as well have more natural test data here.

(I might also play around with some options to just include the test JSON file in the test code. I'll update if I get something that works.)

Reply:
I have added a branch `PC-611-mock-json-example` that has one approach for pulling the JSON out into a file, rather than having it in a C# string.

Reply:
That's much neater. Thanks Jamie! If I understand correctly, the change in the .csproj file:
```
    <ItemGroup>
      <None Update="MockHttpResponses\*">
        <CopyToOutputDirectory>Always</CopyToOutputDirectory>
      </None>
    </ItemGroup>
```
Ensures that the MockHttpResponses are included in the build of the project before the test runs? Or am I misunderstanding.

Reply:
Yes, that's right.

If you check your build output (which is somewhere like `\HerPublicWebsite.UnitTests\bin\Debug\net6.0`), you will see that it includes the `\MockHttpResponses` directory copied there with the JSON file (and it would have any other contents because of the `*`).

This is necessary so that the tests can find the file based on a relative path from where the actual test runner is being executed.